admin/merge_pr

```diff
@@ -0,0 +1,485 @@
+#!/usr/bin/env python
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+# coding: utf-8
+
+"""
+Loop until the checks for a PR are all green, then optionally merge
+the PR.
+"""
+
+from __future__ import print_function
+
+from argparse import ArgumentParser
+from collections import Counter
+from functools import partial
+from itertools import count
+import json
+from operator import itemgetter
+import os
+import sys
+import time
+import urlparse
+
+import requests
+from requests.auth import HTTPBasicAuth
+from twisted.python.constants import Values, ValueConstant
+
+API_BASE_URL = 'https://api.github.com/'
+REPOS_API_PATH = 'repos'
+
+GITHUB_USERNAME_ENV_VAR = 'GITHUB_USER'
+GITHUB_TOKEN_ENV_VAR = 'GITHUB_TOKEN'
+
+JENKINS_USERNAME_ENV_VAR = 'JENKINS_USER'
+JENKINS_PASSWORD_ENV_VAR = 'JENKINS_PASSWORD'
+
+MINIMUM_STATUSES = 50
+MAX_RETRIES = 1
+SLEEP_FOR = 60
+
+GITHUB_VARS_REQUIRED = """"You must set {} and {}.
+You can generate a token at https://github.com/settings/tokens
+(this script requires the repo:status and public_repo scopes)
+""".format(GITHUB_USERNAME_ENV_VAR, GITHUB_TOKEN_ENV_VAR)
+
+JENKINS_VARS_REQUIRED = """Please specify the jenkins user and\
+password in the {} and {} env vars.
+""".format(JENKINS_USERNAME_ENV_VAR, JENKINS_PASSWORD_ENV_VAR)
+
+DESCRIPTION = """Merge a branch when all the tests pass.
+
+Given the URL to a pull request on github this script will
+loop until all the tests pass, and then merge the branch.
+
+It will retry failures on Jenkins, and will also respond
+to any retries you make yourself, and cope with the branch
+being updated while it is running, waiting for the tests
+to pass for the new HEAD.
+
+It can optionally skip the merge if you just want to know
+when all the tests pass.
+
+It requires the following environment variables to be set:
+""" + GITHUB_VARS_REQUIRED + "\n" + JENKINS_VARS_REQUIRED
```

Comment:
This doesn't read properly:
```
(merge-script-FLOC-4013 ? =)[~/.../HybridLogic/flocker]$ ./admin/merge_pr --help
usage: merge_pr [-h] [--no-merge] [--max-retries MAX_RETRIES] url

Merge a branch when all the tests pass. Given the URL to a pull request on
github this script will loop until all the tests pass, and then merge the
branch. It will retry failures on Jenkins, and will also respond to any
retries you make yourself, and cope with the branch being updated while it is
running, waiting for the tests to pass for the new HEAD. It can optionally
skip the merge if you just want to know when all the tests pass. It requires
the following environment variables to be set: "You must set GITHUB_USER and
GITHUB_TOKEN. You can generate a token at https://github.com/settings/tokens
(this script requires the repo:status and public_repo scopes) Please specify
the jenkins user andpassword in the JENKINS_USER and JENKINS_PASSWORD env
vars.
```